day 8

Garret-vin opened the pull request:
user can buy
send email with confirm password

Garret-vin committed:
day 8
user can buy
send email with confirm password

Garret-vin committed:
day 8
user can buy
send email with confirm password
(update 1)

Garret-vin committed:
day 8
user can buy
send email with confirm password

(update 2)
send e-mail in separate thread

Garret-vin committed:
day 8
user can buy
send email with confirm password

(update 2)
send e-mail in separate thread

(update 3)
basket in separate model
added "/admin" to all admin fuctional

Sofasmile commented on src/main/java/com/controller/PaymentServlet.java:
```diff
@@ -0,0 +1,41 @@
+        User user = (User) session.getAttribute("user");
+        Code code = new Code(user);
+
+        new Thread(() -> mailService.sendConfirmCode(code)).start();
```
why we use multithreading here?

Sofasmile commented on src/main/java/com/model/Code.java:
```diff
@@ -0,0 +1,30 @@
+
+public class Code {
+
+    private User user;
```
maybe better add Code code in user model?
this class shouldn't depend from user

Sofasmile commented on src/main/webapp/payment.jsp:
```diff
@@ -0,0 +1,15 @@
+<h1>Создание заказа</h1>
+<form action="/payment" method="post">
+    E-mail <input type="text" name="email" value="${userEmail}"><br>
+    Номер телефона <input type="tel" value="+123456789012"><br>
```
where we work with this data?
better create separate entity - Order and work with them

Garret-vin committed:
day 8
user can buy
send email with confirm password

(update 4)
add class order
basket in separate class
add method update

Garret-vin committed:
day 8
user can buy
send email with confirm password

(update 4)
add class order
basket in separate class
add method update

(update 5)
add CodeGeneratorUtil

Garret-vin committed:
day 8
user can buy
send email with confirm password

(update 4)
add class order
basket in separate class
add method update

(update 5)
add CodeGeneratorUtil

Sofasmile reviewed (commented):
we need to create classes at service and dao layer for Order

Garret-vin committed:
day 8
user can buy
send email with confirm password

(update 6)
Order dao+service layer

Sofasmile commented on src/main/java/com/controller/UserBuyServlet.java:
```diff
@@ -0,0 +1,32 @@
+        Basket basket = (Basket) req.getSession().getAttribute("basket");
+
+        Optional<Product> optionalProduct = productService.getById(productId);
+        optionalProduct.ifPresent(basket::addProduct);
```
we don't need to create method addProduct() in model
all methods need to be in dao and service layers

Garret-vin committed:
day 8
user can buy
send email with confirm password

(update 7)
add basket dao
check code from db(it was from session)
